src/ui/redux/actions/publish.js

```diff
@@ -69,9 +69,18 @@ export const doUpdatePublishForm = (publishFormValue: UpdatePublishFormData) =>
   });
 
 export const doUploadThumbnail = (filePath: string) => (dispatch: Dispatch) => {
-  const thumbnail = fs.readFileSync(filePath);
-  const fileExt = path.extname(filePath);
-  const fileName = path.basename(filePath);
+  // Assuming filePath is a bufferArray
+  let thumbnail = filePath;
+  let fileExt = '.png';
+  let fileName = 'thumbnail.png';
+  let fileType = 'image/png';
+
+  if (typeof filePath === 'string') {
```

Comment:
We shouldn't need to do this because we've declared that `filePath` will be a string. Maybe we should add a second argument for a buffer?

`export const doUploadThumbnail = (filePath?: string, thumbnailBuffer: Buffer)`

(I'm not sure what that flow type should be)

Reply:
I did as you said, but do you think wrapping them two in one object as an argument is any better?

Reply:
I think two arguments is fine.